logging/telemetryConfig.go

```diff
@@ -120,6 +131,26 @@ func (cfg TelemetryConfig) getInstanceName() string {
 	return fmt.Sprintf("%s:", pathHashStr[:16])
 }
 
+// SanitizeTelemetryString applies sanitization rules and returns the sanitized string.
+func SanitizeTelemetryString(input string, maxParts int) string {
+	// Truncate to a reasonable size.
+	input = input[:maxParts * 100]
+
+	// Limit to alphanumeric, and _-
+	if reg != nil {
+		input = reg.ReplaceAllString(input, "")
+	}
+
+	parts := strings.SplitN(input, ":", maxParts)
```

Comment:
are you trying to split `host:port` ? IPv6 addresses have ':' in them. maybe `net.SplitHostPort()` ?

Reply:
The main reason for this is the "TelemetryHost" field is in a `GUID:Name` format, and I wanted to apply a length threshold to each separately. For sanitizing `AddressHeader` maxParts is 1, so  it should be unchanged.

Reply:
Added an ipv6 test case